kuzzleio/kuzzle
# Framework: BackendImport

Closes #2077

## What does this PR do ?

Add a new `BackendImport` class accessible from `backend.import`.
This class offers 5 new methods:

- `mappings` Import indexes and their collections mappings (and settings)
- `profiles` Import profiles
- `roles` Import roles
- `userMappings` Import user mappings
- `users` Import users. If they already exist in DB at starting phase, we can `skip` their importation or `overwrite` DB to prevent any change on these users. 

All of these methods are __idempotent__ (can be call multiples times). Also, the __call order doesn't matter__ because it's only when the app starts that import will be made (in the right order) in the database. Finally, all of these methods except `userMappings` come with __merge ability__ when called multiple times (old and new profiles, roles, users will be combined as long as collections contained in same index).

### Changes related to functional tests 

- From now, tests order matters. BackendImport tests need to be executed before resetting database (hence **#** in `#BackendImport.feature`)
- `docker/script/start-kuzzle-dev`: `backend._support` has been replaced by multiple `backend.import` which means dev environment is slightly different.
- Deprecation warnings are removed for all tests

### How should this be manually tested?

- Clone Kuzzle
- Import some mappings
- Start Kuzzle and see the result on the admin console
